backend/tracim_backend/lib/core/event.py

```diff
@@ -98,19 +99,43 @@ class EventBuilder:
 
     def __init__(self, config: CFG) -> None:
         self._config = config
-        self._current_user = None  # type: typing.Optional[User]
+        # NOTE S.G. 2020-06-04: need thread local storage as EventBuilder()
+        # is created only once/wsgi application
+        self._thread_local = threading.local()
+        self._thread_local.current_user = None
+        self._thread_local.pending_events = []
+
+    @property
+    def _current_user(self) -> typing.Optional[User]:
+        try:
+            return self._thread_local.current_user
+        except AttributeError:
+            return None
 
     @hookimpl
     def on_current_user_set(self, user: User) -> None:
-        self._current_user = user
+        self._thread_local.current_user = user
 
     @hookimpl
     def on_request_session_created(self, request: TracimRequest, session: TracimSession) -> None:
-        event.listen(session, "pending_to_persistent", self._publish_event)
+        commit_event = "after_flush"
+        if self._config.JOBS__PROCESSING_MODE == self._config.CST.ASYNC:
+            # We need after commit event when processing in async
```

Comment:
Can't we do both sync and async with "after commit" ?

Reply:
Unfortunately not as the session is un-useable `after_commit` (I tried).